packages/syntax/src/renderingPlugin.tsx

```diff
@@ -57,32 +58,94 @@ export function processToken(token: TreeItem, key?: number): React.ReactNode {
 }
 
 const cellStyle = css`
-  user-select: none;
   border-spacing: 0;
   padding: 0;
   vertical-align: top;
+
+  &:first-of-type {
+    padding-left: 12px;
+  }
+
+  &:last-of-type {
+    padding-right: 12px;
+  }
 `;
 
-export const numberCellDataProp = createDataProp('code-number-cell');
+function getHighlightedRowStyle(darkMode: boolean) {
+  let backgroundColor: string, backgroundImage: string, borderColor: string;
+
+  if (darkMode) {
+    backgroundColor = 'transparent';
+    backgroundImage = `linear-gradient(90deg, ${uiColors.gray.dark3}, transparent)`;
+    borderColor = uiColors.gray.dark3;
+  } else {
+    backgroundColor = uiColors.yellow.light3;
+    backgroundImage = 'none';
+    borderColor = uiColors.yellow.light2;
+  }
+
+  return css`
+    background-color: ${backgroundColor};
+    background-image: ${backgroundImage};
+
+    // Selects all children of a highlighted row, and adds a border top
+    & > td {
+      border-top: 1px solid ${borderColor};
+    }
+
+    // Selects following rows after a highlighted row, and adds a border top
+    // We don't add border bottoms here to support consecutive highlighted rows.
+    & + tr > td {
+      border-top: 1px solid ${borderColor};
+    }
+
+    // Remove borders between consecutive highlighted rows
+    & + & > td {
+      border-top: 0;
+    }
+
+    // If the highlighted row is the last child, then we add a border bottom
+    &:last-child > td {
+      border-bottom: 1px solid ${borderColor};
+    }
+  `;
+}
 
 interface LineTableRowProps {
-  lineNumber: number;
+  lineNumber?: number;
   children: React.ReactNode;
+  highlighted?: boolean;
+  darkMode: boolean;
 }
 
-export function LineTableRow({ lineNumber, children }: LineTableRowProps) {
-  const numberCellStyle = css`
-    padding-right: 24px;
-  `;
+export function LineTableRow({
+  lineNumber,
+  highlighted,
+  darkMode,
+  children,
+}: LineTableRowProps) {
+  const numberColor = uiColors.gray[darkMode ? 'dark1' : 'light1'];
+  const highlightedNumberColor = darkMode
+    ? uiColors.gray.light2
+    : uiColors.yellow.dark2;
 
   return (
-    <tr>
-      <td
-        {...numberCellDataProp.prop}
-        className={cx(cellStyle, numberCellStyle)}
-      >
-        {lineNumber}
-      </td>
+    <tr className={cx({ [getHighlightedRowStyle(darkMode)]: highlighted })}>
```

Comment:
would we benefit from memoizing `getHighlightedRowStyle`?

Reply:
Don't think so – emotion does a lot of caching for these values, and I believe there's a chance that the emotion className might change without a dependency of `useMemo` changing.